crates/uv-cli/src/lib.rs

```diff
@@ -235,6 +235,14 @@ pub struct GlobalArgs {
     /// Change to the given directory prior to running the command.
     #[arg(global = true, long, hide = true)]
     pub directory: Option<PathBuf>,
+
+    /// Run the command within the given project.
+    ///
+    /// All `pyproject.toml`, `uv.toml`, and `.python-version` files will be discovered by walking
+    /// up the directory tree from the project root, while relative paths will be resolved relative
+    /// to the current working directory.
+    #[arg(global = true, long)]
+    pub project: Option<PathBuf>,
```

Comment:
Ignored in the `uv pip` interface though?

Should we note this affects project environment discovery too?